### PR TITLE
feat: Add cli support for additional linux based architectures, s390x + ppc64le

### PR DESCRIPTION
**Pull request by davidkarlsen**
While AIX isn't a very popular platform these days, it would be valuable for some legacy platforms to be able to integrate with argo-cd via the CLI in order to die in a nice way :)

Signed-off-by: David J. M. Karlsen <david@davidkarlsen.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [X] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [X] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [X] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -234,6 +234,7 @@ release-cli: clean-debug build-ui
 	make BIN_NAME=argocd-linux-amd64 GOOS=linux argocd-all
 	make BIN_NAME=argocd-linux-arm64 GOOS=linux GOARCH=arm64 argocd-all
 	make BIN_NAME=argocd-windows-amd64.exe GOOS=windows argocd-all
+	make BIN_NAME=argocd-aix-ppc64 GOOS=aix GOARCH=ppc64 argocd-all
```
Comment (krishvoor) [inline Makefile:237]: Can you consider adding these to your PR?

```
make BIN_NAME=argocd-linux-ppc64le GOOS=linux GOARCH=ppc64le argocd-all
make BIN_NAME=argocd-linux-s390x GOOS=linux GOARCH=s390x argocd-all
```

Comment (davidkarlsen) [reply Makefile:239]: @krishvoor fixed

**Review by krishvoor — COMMENTED** [review]
Thanks for the changes, lgtm!

**Comment by krishvoor** [comment]
**Minor nitpick:**
Looks like Semantic Pull Request is throwing a red-flag

Can you update your PR Title - something similar to [this](https://github.com/zeke/semantic-pull-requests#example-scenario) like
`chore: Add cli support for additional architectures and aix`

**Comment by davidkarlsen** [comment]
@krishvoor fixed

**Comment by krishvoor** [comment]
Hi @rishabh625  - can you review it?

**Review by rishabh625 — APPROVED** [review]
LGTM!

**Comment by davidkarlsen** [comment]
Hmm, CI does not actually build that steps it seems.
It fails when I:
```shell
make BIN_NAME=argocd-aix-ppc64 GOOS=aix GOARCH=ppc64 argocd-all
find /Users/et2448/projects/ext/github.com/argo-cd -name debug.test | xargs rm -f
CGO_ENABLED=0 GOOS=aix GOARCH=ppc64 go build -v -ldflags '-X github.com/argoproj/argo-cd/v2/common.version=2.4.0 -X github.com/argoproj/argo-cd/v2/common.buildDate=2022-04-05T20:47:18Z -X github.com/argoproj/argo-cd/v2/common.gitCommit=f8c4d217615efee278b9e32f1f5dd5e0d3b3d9ee -X github.com/argoproj/argo-cd/v2/common.gitTreeState=clean -X github.com/argoproj/argo-cd/v2/common.kubectlVersion=v0.23.1 -extldflags "-static"' -o /Users/et2448/projects/ext/github.com/argo-cd/dist/argocd-aix-ppc64 ./cmd
package github.com/argoproj/argo-cd/v2/cmd
        imports github.com/argoproj/argo-cd/v2/cmd/argocd-application-controller/commands
        imports github.com/argoproj/pkg/stats: build constraints exclude all Go files in /Users/et2448/go/pkg/mod/github.com/argoproj/pkg@v0.11.1-0.20211203175135-36c59d8fafe0/stats
make: *** [argocd-all] Error 1
```
locally

I guess not all packages can be cross-compiled

**Comment by krishvoor** [comment]
Right, CI might need QEMU Emulation to get it working..
Also, we have a patch for ppc64le/s390x for build-time/test dependencies [here](https://github.com/argoproj/argo-cd/pull/8890)

**Comment by davidkarlsen** [comment]
> Right, CI might need QEMU Emulation to get it working.. Also, we have a patch for ppc64le/s390x for build-time/test dependencies [here](https://github.com/argoproj/argo-cd/pull/8890)

I guess it would be good to build all architectures in CI, so one does not get a surprise at release-time.

**Comment by 34fathombelow** [comment]
> Hmm, CI does not actually build that steps it seems. It fails when I:
> 
> ```shell
> make BIN_NAME=argocd-aix-ppc64 GOOS=aix GOARCH=ppc64 argocd-all
> find /Users/et2448/projects/ext/github.com/argo-cd -name debug.test | xargs rm -f
> CGO_ENABLED=0 GOOS=aix GOARCH=ppc64 go build -v -ldflags '-X github.com/argoproj/argo-cd/v2/common.version=2.4.0 -X github.com/argoproj/argo-cd/v2/common.buildDate=2022-04-05T20:47:18Z -X github.com/argoproj/argo-cd/v2/common.gitCommit=f8c4d217615efee278b9e32f1f5dd5e0d3b3d9ee -X github.com/argoproj/argo-cd/v2/common.gitTreeState=clean -X github.com/argoproj/argo-cd/v2/common.kubectlVersion=v0.23.1 -extldflags "-static"' -o /Users/et2448/projects/ext/github.com/argo-cd/dist/argocd-aix-ppc64 ./cmd
> package github.com/argoproj/argo-cd/v2/cmd
>         imports github.com/argoproj/argo-cd/v2/cmd/argocd-application-controller/commands
>         imports github.com/argoproj/pkg/stats: build constraints exclude all Go files in /Users/et2448/go/pkg/mod/github.com/argoproj/pkg@v0.11.1-0.20211203175135-36c59d8fafe0/stats
> make: *** [argocd-all] Error 1
> ```
> 
> locally
> 
> I guess not all packages can be cross-compiled

I also get the same error running this locally in Linux with QEMU.  This would definitely cause a release to fail.  @davidkarlsen Would you consider removing `make BIN_NAME=argocd-aix-ppc64 GOOS=aix GOARCH=ppc64 argocd-all`?  The other additions would be really helpful since 2.4 will have support for `ppc64le` & `s390x`.

**Comment by davidkarlsen** [comment]
@34fathombelow fixed

**Comment by 34fathombelow** [comment]
@davidkarlsen  lgtm!  If you don't mind changing the title of the PR to reflect only the architectures added.

**Review by crenshaw-dev — APPROVED** [review]
lgtm. @alexmt  any objections?

**Pull request merged**